docker-java-transport-okhttp/src/main/java/com/github/dockerjava/okhttp/OkDockerHttpClient.java

```diff
@@ -0,0 +1,284 @@
+package com.github.dockerjava.okhttp;
+
+import com.github.dockerjava.core.DockerClientConfig;
+import com.github.dockerjava.core.DockerHttpClient;
+import com.github.dockerjava.core.SSLConfig;
+import okhttp3.ConnectionPool;
+import okhttp3.Dns;
+import okhttp3.HttpUrl;
+import okhttp3.MediaType;
+import okhttp3.OkHttpClient;
+import okhttp3.RequestBody;
+import okhttp3.ResponseBody;
+import okio.BufferedSink;
+import okio.Okio;
+
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.X509TrustManager;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.UncheckedIOException;
+import java.net.InetAddress;
+import java.net.URI;
+import java.security.cert.X509Certificate;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+public final class OkDockerHttpClient implements DockerHttpClient {
+
+    public static final class Factory {
+
+        private DockerClientConfig dockerClientConfig = null;
+
+        private Integer readTimeout = null;
+
+        private Integer connectTimeout = null;
+
+        private Boolean retryOnConnectionFailure = null;
```

Comment:
why don't we have defaults here?

Reply:
default is "unspecified" (`null`) which means that it is up to the http library we use to decide on them

Reply:
Also note that I added it to keep backward compatibility with `AbstractDockerCmdFactory`. I wish I could drop them since I believe they are not very helpful